tests/test_basic.py

```diff
@@ -104,50 +104,46 @@ def sub():
     assert result.output == "sub"
 
 
-def test_basic_option(runner):
+@pytest.mark.parametrize(
+    ("values", "expect"),
+    [
+        ("--foo=no value", "FOO:[no value]"),
+        ("--foo=42", "FOO:[42]"),
+        ("--foo", "Option '--foo' requires an argument."),
+        ("--foo=", "FOO:[]"),
+        ("--foo=\N{SNOWMAN}", "FOO:[\N{SNOWMAN}]"),
+    ],
+)
+def test_basic_option(runner, values, expect):
     @click.command()
     @click.option("--foo", default="no value")
     def cli(foo):
         click.echo(f"FOO:[{foo}]")
 
-    result = runner.invoke(cli, [])
-    assert not result.exception
-    assert "FOO:[no value]" in result.output
-
-    result = runner.invoke(cli, ["--foo=42"])
-    assert not result.exception
-    assert "FOO:[42]" in result.output
-
-    result = runner.invoke(cli, ["--foo"])
-    assert result.exception
-    assert "Option '--foo' requires an argument." in result.output
-
-    result = runner.invoke(cli, ["--foo="])
-    assert not result.exception
-    assert "FOO:[]" in result.output
+    result = runner.invoke(cli, [values])
 
-    result = runner.invoke(cli, ["--foo=\N{SNOWMAN}"])
-    assert not result.exception
-    assert "FOO:[\N{SNOWMAN}]" in result.output
+    assert result.exception is None
+    assert result.return_value == expect
```

Comment:
`return_value` is only set when running in `standalone_mode=False`, you can test `result.output` in this case.